lib/df-detection.c

```diff
@@ -40,57 +40,66 @@ static void detect_intersection(void *kernel_addr)
 {
 	int i;
 
-	for (i = 0; i < current->num_read; i++) {
-		if (!is_intersect(current->dfetch_addresses[i],
-				  current->dfetch_addresses[current->num_read],
-				  kernel_addr))
+	for (i = 0; i < current->dfetch.num_read; i++) {
+		if (!is_intersect(
+			current->dfetch.dfetch_addresses[i],
```

Comment:
This code is using current->dfetch a lot, so it's better to declare a pointer to that struct and use it instead.
The compiler will do that anyway, but the code will also become cleaner.